include/swift/AST/AutoDiff.h

```diff
@@ -32,26 +32,36 @@ enum class DifferentiabilityKind: uint8_t {
 
 class ParsedAutoDiffParameter {
 public:
-  enum class Kind { Named, Self };
+  enum class Kind { Named, Ordered, Self };
 
 private:
   SourceLoc Loc;
   Kind Kind;
   union Value {
-    struct { Identifier Name; }; // Index
+    struct { Identifier Name; }; // Named
+    struct { unsigned int Index; }; // Ordered
     struct {};                  // Self
     Value(Identifier name) : Name(name) {}
+    Value(unsigned int index) : Index(index) {}
     Value() {}
   } V;
 
 public:
   ParsedAutoDiffParameter(SourceLoc loc, enum Kind kind, Value value)
     : Loc(loc), Kind(kind), V(value) {}
+  
+  ParsedAutoDiffParameter(SourceLoc loc, enum Kind kind, unsigned int order)
+  : Loc(loc), Kind(kind), V(order) {}
 
   static ParsedAutoDiffParameter getNamedParameter(SourceLoc loc,
                                                    Identifier name) {
     return { loc, Kind::Named, name };
   }
+  
+  static ParsedAutoDiffParameter getOrderedParameter(SourceLoc loc,
+                                                   unsigned int order) {
```

Comment:
```suggestion
                                                     unsigned int index) {
```
Align argument. `order` -> `index`.